cassiopeia/data/CassiopeiaTree.py

```diff
@@ -1334,14 +1651,130 @@ def filter_nodes(self, condition: Callable[[str], bool]) -> List[str]:
-            pairs: Pairs of nodes for which to find LCAs
+            pairs: Pairs of nodes for which to find LCAs. If not provided, LCAs of
+                all pairs are computed. Defaults to None.
 
         Returns:
             A generator of ((u, v), LCA) tuples.
+
+        Raises:
+            CassiopeiaTreeError if the tree has not been initialized.
+        """
+        self.__check_network_initialized()
+        return nx.tree_all_pairs_lowest_common_ancestor(
+            self.__network, root=self.root, pairs=pairs
+        )
+
+    def find_lca(self, *nodes: str) -> str:
+        """Finds the LCA of all provided nodes.
+
+        Internally, this function calls :func:`get_all_ancestors` for each provided
+        node and finds the deepest node that is shared.
+
+        Args:
+            *nodes: Nodes for which to find the LCA. At least two must be provided.
+
+        Returns:
+            The LCA node
+
+        Raises:
+            CassiopeiaTreeError if less than two nodes were provided or if the
+            tree has not been initialized.
         """
         self.__check_network_initialized()
-        return nx.tree_all_pairs_lowest_common_ancestor(self.__network, root=self.root, pairs=pairs)
+        nodes = set(nodes)
+        if len(nodes) < 2:
+            raise CassiopeiaTreeError(
+                "At least two distinct nodes must be provided"
+            )
+
+        # Reversing get_all_ancestors gives a list of nodes from the root to each
+        # node. Since get_all_ancestors doesn't include the node itself, we add
+        # those manually.
+        all_ancestors = [
+            reversed(self.get_all_ancestors(node, include_node=True))
+            for node in nodes
+        ]
+        last_ancestor = self.root
+        for ancestors in zip(*all_ancestors):
+            if len(set(ancestors)) > 1:
+                break
+            last_ancestor = ancestors[0]
+        return last_ancestor
+
+    def get_distance(self, node1: str, node2: str) -> float:
+        """Compute the branch distance between two nodes.
+
+        Internally, this function converts the tree to an undirected graph and
+        finds the shortest path by calling :func:`nx.shortest_path_length`.
+
+        Args:
+            node1: First node
+            node2: Second node
+
+        Returns:
+            The branch distance between the two nodes
+
+        Raises:
+            CassiopeiaTreeError if the tree has not been initialized.
+        """
+        self.__check_network_initialized()
+
+        key = frozenset({node1, node2})
+        if (
+            "distances" not in self.__cache
+            or key not in self.__cache["distances"]
+        ):
+            distances = self.__cache.setdefault("distances", {})
+            undirected_network = self.__network.to_undirected()
+            distances[key] = nx.shortest_path_length(
+                undirected_network, source=node1, target=node2, weight="time"
+            )
```

Comment:
"time" is used for nodes, the attribute for branch length is "length". I tested it really quickly and this just returns the unweighted path length (i.e. number of seperating) for nodes. Key to fix this or else the sampling in the merging step will not be correct!

Reply:
Oof. Thanks for catching this!
Thankfully for my simulations I used an older implementation of this where I manually calculate the branch distance (instead of using networkx), so they should be fine.